src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/CreateCitizenClaimCallbackHandlerTest.java

```diff
@@ -0,0 +1,192 @@
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.util.Collections;
+
+import static java.math.BigDecimal.TEN;
+import static java.time.LocalDate.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.cmc.ccd.domain.CaseEvent.CREATE_CASE;
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.Role.CITIZEN;
+import static uk.gov.hmcts.cmc.domain.models.PaymentStatus.SUCCESS;
+import static uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaim.withFullClaimData;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CreateCitizenClaimCallbackHandlerTest {
+
+    public static final String REFERENCE_NO = "000MC001";
+    public static final LocalDate ISSUE_DATE = now();
+    public static final LocalDate RESPONSE_DEADLINE = ISSUE_DATE.plusDays(14);
+    private static final String BEARER_TOKEN = "Bearer let me in";
+    private static final String NEXT_URL = "http://nexturl.test";
+
+    @Mock
+    private CaseDetailsConverter caseDetailsConverter;
+
+    @Mock
+    private IssueDateCalculator issueDateCalculator;
+
+    @Mock
+    private ReferenceNumberRepository referenceNumberRepository;
+
+    @Mock
+    private ResponseDeadlineCalculator responseDeadlineCalculator;
+
+    @Mock
+    private CaseMapper caseMapper;
+
+    @Mock
+    private PaymentsService paymentsService;
+
+    @Captor
+    private ArgumentCaptor<Claim> claimArgumentCaptor;
+
+    private CallbackParams callbackParams;
+    private CallbackRequest callbackRequest;
+    private CreateCitizenClaimCallbackHandler createCitizenClaimCallbackHandler;
+
+    private CaseDetails caseDetails = CaseDetails.builder().id(3L).data(Collections.emptyMap()).build();
+
+    @Before
+    public void setUp() {
+        createCitizenClaimCallbackHandler =
+            new CreateCitizenClaimCallbackHandler(
+            caseDetailsConverter,
+            issueDateCalculator,
+            referenceNumberRepository,
+            responseDeadlineCalculator,
+            caseMapper,
+            paymentsService
+        );
+
+        callbackRequest = CallbackRequest
+            .builder()
+            .eventId(CREATE_CASE.getValue())
+            .caseDetails(caseDetails)
+            .build();
+
+        Mockito.when(issueDateCalculator.calculateIssueDay(any())).thenReturn(ISSUE_DATE);
+        Mockito.when(responseDeadlineCalculator.calculateResponseDeadline(ISSUE_DATE)).thenReturn(RESPONSE_DEADLINE);
+        Mockito.when(referenceNumberRepository.getReferenceNumberForCitizen()).thenReturn(REFERENCE_NO);
+    }
+
+    @Test
+    public void shouldSuccessfullyReturnCallBackResponseWhenSuccessfulPayment() {
+
+        OffsetDateTime paymentDate = OffsetDateTime.parse("2017-02-03T10:15:30+01:00");
```

Comment:
Was this in France? UK wasn't in +01:00 on this date 😝

Reply:
i copied that from somewhere :). updated to use now().